CONTRIBUTING.md

```diff
@@ -0,0 +1,190 @@
+
+Simple rules for table-driven tests:
+
+1. Keep tests independent (ordering of tests should not impact testing);
+2. Test one construct or aspect per test;
+3. Every test should make it obvious what is being tested;
+
+## 5. Pull Requests
+
+* All pull requests must target `main`;
+* Updates to __release branches__ are cherry-picked from `main`;
+
+Requirements to be merged:
+
+* The source branch must be __up-to-date__ with `main`;
+* All __commits__ must be __signed__;
+* All GitHub __action checks must pass__;
+* __Documentation and Specs__ must be __updated__ accordingly;
+* PR must be __reviewed and approved__;
+
+### PR Title & Description
+
+Title format: `<impact>: <issue #> - <short description>`
+
+Impact:
+
+- _Breaking_ (major version bump)
+- _Feature_ (minor version bump)
+- _Fix_ (patch version bump)
+
+Other acceptable impact indicators are:
+
+- _Docs_
+- _Tests_
+
+The PR __title__ is more than a mere commit message. It should convey something of substantial significance... A review of the repository pull request titles should provide a very clear progression path, that makes sense to someone visiting the repository for the first time.
+
+Provide a clear and concise __description__ of what has changed and why. Use full sentences with appropriate punctuation. It should be a _summary of changes_, taken from the major commits that make up the PR.
+
+Do add references to any applicable issues at the very end using the notation: `Fixes #123`.
+
+Example:
+
+```
+Docs: #123 - Updates & Quality Control
+
+Add contribution guidelines to provide an overview of the processes and policies to ensure quality collaboration and coordination within the repository.
+
+Add security policy to detail the processes for reporting security issues and vulnerability disclosure.
+
+Fixes #123, #125, #127
+```
+
+### Commits & Commit Messages
+
+Make smaller commits more often. This allows the commit log to tell the story of what has changed, and why.
+
+The __first line__ of every commit message must complete the following sentence: _"This commit will..."_ (use imperative mood);
+
+An optional __main content__ (separated from the first line with a blank line) should describe or elaborate on what has changed and why. Use full sentences with appropriate punctuation.
+
+Reference issues at the very end using the notation: `Fixes #123`.
+
+Example:
+
+```
+Add contribution guidelines
+
+Contribution guidelines provide an overview of the processes and policies to ensure quality collaboration and coordination within the repository.
+
+Fixes #125
+```
+
+See [How to Write a Git Commit Message](https://cbea.ms/git-commit/) for more details.
+
+## 6. Review
+
+Verify PR Requirements:
+
+- PR title is appropriate;
+- PR description is sufficient;
+- Branch-name is appropriate;
+- PR target branch is `main`;
+- Source is up-to-date with `main`;
+- All commits are signed;
+- All GitHub action checks pass;
+- Documentation and Specs are updated appropriately;
```

Comment:
We can later include a GitHub workflow automation to ensure this is applied in every new PR.